extensions/smallrye-health/deployment/src/main/java/io/quarkus/smallrye/health/deployment/SmallRyeHealthProcessor.java

```diff
@@ -278,4 +326,162 @@ public void transform(TransformationContext ctx) {
 
         });
     }
+
+    // UI
+
+    @BuildStep
+    @Record(ExecutionTime.STATIC_INIT)
+    void registerUiExtension(
+            BuildProducer<RouteBuildItem> routeProducer,
+            BuildProducer<GeneratedResourceBuildItem> generatedResourceProducer,
+            BuildProducer<NativeImageResourceBuildItem> nativeImageResourceProducer,
+            BuildProducer<NotFoundPageDisplayableEndpointBuildItem> notFoundPageDisplayableEndpointProducer,
+            SmallRyeHealthRecorder recorder,
+            LaunchModeBuildItem launchMode,
+            LiveReloadBuildItem liveReload,
+            HttpRootPathBuildItem httpRootPath,
+            CurateOutcomeBuildItem curateOutcomeBuildItem) throws Exception {
+
+        if (!health.enableUi) {
+            return;
+        }
+        if ("/".equals(health.rootPathUi)) {
+            throw new ConfigurationError(
+                    "quarkus.smallrye-health.root-path-ui was set to \"/\", this is not allowed as it blocks the application from serving anything else.");
+        }
+
+        String healthPath = httpRootPath.adjustPath(health.rootPath);
+
+        if (launchMode.getLaunchMode().isDevOrTest()) {
+            CachedHealthUI cached = liveReload.getContextObject(CachedHealthUI.class);
+            boolean extractionNeeded = cached == null;
+
+            if (cached != null && !cached.cachedHealthPath.equals(healthPath)) {
```

Comment:
IIUC, the UI artifacts are only static files so why do we need to reextract them when the path to UI changes? Can't you just reuse already created temp dir and just change the path in the cached instance?

Reply:
Yea, the idea is actually that you can change the `/health` path and the UI sould pick this up. But that is not yet implemented, as the user can change that path the settings, but maybe I should just implement it now.

Reply:
Point is, then it's not static anymore...

Reply:
Ok, I now added the dynamic part.